src/data/polynomial/basic.lean

```diff
@@ -80,12 +79,12 @@ by rw [mul_assoc, X_pow_mul, ←mul_assoc]
 lemma commute_X (p : polynomial R) : commute X p := X_mul
 
 /-- coeff p n is the coefficient of X^n in p -/
-def coeff (p : polynomial R) := p.to_fun
+def coeff (p : polynomial R) : ℕ → R := @coe_fn (ℕ →₀ R) _ p
```

Comment:
This gives slightly better behavior for unfolding `coeff`, I think - but isn't actually needed for the rest of the patch